arelle/ModelValue.py

```diff
@@ -122,80 +124,82 @@ def qnameEltPfxName(element, prefixedName, prefixException=None):
         else:
             return self.localName
-    def __eq__(self,other):
-        try:
-            return (self.qnameValueHash == other.qnameValueHash and 
+
+    def __eq__(self, other):
+        if isinstance(other, QName):
+            return (self.qnameValueHash == other.qnameValueHash and
                     self.localName == other.localName and self.namespaceURI == other.namespaceURI)
-        except AttributeError:
-            return False
-        ''' don't think this is used any longer
-        if isinstance(other,_STR_BASE):
-            # only compare nsnames {namespace}localname format, if other has same hash
-            return self.__hash__() == other.__hash__() and self.clarkNotation == other
-        elif isinstance(other,QName):
-            return self.qnameValueHash == other.qnameValueHash and \
-                    self.namespaceURI == other.namespaceURI and self.localName == other.localName
-        elif isinstance(other,ModelObject):
-            return self.namespaceURI == other.namespaceURI and self.localName == other.localName
-        '''
-        '''
-        try:
-            return (self.qnameValueHash == other.qnameValueHash and 
-                    self.namespaceURI == other.namespaceURI and self.localName == other.localName)
-        except AttributeError:  # other may be a model object and not a QName
-            try:
-                return self.namespaceURI == other.namespaceURI and self.localName == other.localName
-            except AttributeError:
-                return False
         return False
-        '''
-    def __ne__(self,other):
-        return not self.__eq__(other)
-    def __lt__(self,other):
-        return (self.namespaceURI is None and other.namespaceURI) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI < other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName < other.localName)
-    def __le__(self,other):
-        return (self.namespaceURI is None and other.namespaceURI) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI < other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName <= other.localName)
-    def __gt__(self,other):
-        return (self.namespaceURI and other.namespaceURI is None) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI > other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName > other.localName)
-    def __ge__(self,other):
-        return (self.namespaceURI and other.namespaceURI is None) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI > other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName >= other.localName)
+
+    def __lt__(self, other):
+        if isinstance(other, QName):
```

Comment:
I'm a little worried that these changes in duck typing behavior might break someone who's integrating